compiler/rustc_trait_selection/src/error_reporting/traits/mod.rs

```diff
@@ -163,14 +163,26 @@ impl<'a, 'tcx> TypeErrCtxt<'a, 'tcx> {
         // Ensure `T: Sized` and `T: WF` obligations come last. This lets us display diagnostics
         // with more relevant type information and hide redundant E0282 errors.
         errors.sort_by_key(|e| match e.obligation.predicate.kind().skip_binder() {
+            ty::PredicateKind::Subtype(_)
+                if e.obligation.cause.span.macro_backtrace().next().is_some_and(
```

Comment:
I think this can be simplified with some other method like `span.parent` or `span.expn_kind` or something? Look on the span type for relevant functions, b/c using `backtrace.next` smells wrong 🤔